### PR TITLE
FKVR-26: Experience Editing

### PR DESCRIPTION
**Pull request by patrickocoffeyo**
## [FKVR-26](https://fourkitchens.atlassian.net/browse/FKED-26)

**This PR does the following:**
- Adds API calls, actions, reducers, etc to support experience edit API calls.
- Adjusts `ExperienceForm` so that it can handle both creating and editing experiences.
- Fixes a small issue with the API lib.

### To Review:

- [x] Checkout this branch.
- [x] Run `yarn start`.
- [x] Log into EditVR.
- [x] Note that you can open an experience, edit it's details, and save it.

### CODE REVIEW

File: src/components/ExperienceForm/ExperienceForm.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          }
+          className={classes.textField}
+        />
+        <Button
+          variant="raised"
+          color="primary"
+          type="submit"
+          className={classes.button}
+        >
+          {experienceSlug ? 'Save' : 'Create'}
```
Comment (techninja) [inline src/components/ExperienceForm/ExperienceForm.js]: I feel like we could get away with this being being "Save" or "Save Experience" for both.

**Comment by techninja** [comment]
I note that you are prevented from saving an edited experience if it doesn't pass validation (like the Blanton demo on Dev). These messages from 422 style errors should probably be surfaced to users beyond simply "There was an error".

File: src/components/ExperienceForm/ExperienceForm.js (hunk 1 above)
Comment (patrickocoffeyo) [reply src/components/ExperienceForm/ExperienceForm.js]: That's a good point, I think "Save" would be fine :100:

File: src/actions/user.test.js [hunk 2]
```diff
@@ -50,6 +51,7 @@ describe('actions->user', () => {
         }
       })
       .next()
+      .next()
```
[the remark this answers is not in the document]
Comment (patrickocoffeyo) [reply src/actions/user.test.js]: Yeah, that is a good point. I'll insert a short comment describing why we need that second `next` call.

**Comment by patrickocoffeyo** [comment]
@techninja Good call, I added case for handling 422 responses... I think there's something funky with the blanton experience: `field_scenes.10.target_id: The referenced entity (node: 245) does not exist`. I think this is one of those cases where something is borked on the API side of things... we likely need to log errors like these somewhere so we can collect them and respond to issues. I've been considering implementing some sort of front end logging, and I think this refactor is a great opportunity to do so.

**Comment by techninja** [comment]
Awesome work! Didn't expect a number of these improvements but they're really worth it. Now if only Drupal would behave better about being flogged like the datastore we want it to be 😆. Apparently the next version of Drupal will be WAY better at this.

**Comment by patrickocoffeyo** [comment]
Haha, well I'm cautiously hopeful about the next version of Drupal. We'll see :dancer:

**Pull request merged**